drv/lpc55-update-server/src/images.rs

```diff
@@ -26,79 +45,494 @@ extern "C" {
 // Location of the NXP header
 pub const HEADER_BLOCK: usize = 0;
 
-// NXP LPC55's mixed header/vector table offsets
-const RESET_VECTOR_OFFSET: usize = 0x04;
-pub const LENGTH_OFFSET: usize = 0x20;
-pub const HEADER_OFFSET: u32 = 0x130;
-const MAGIC_OFFSET: usize = HEADER_OFFSET as usize;
+// An image may have an ImageHeader located after the
+// LPC55's mixed header/vector table.
+pub const IMAGE_HEADER_OFFSET: u32 = 0x130;
 
-// Perform some sanity checking on the header block.
-pub fn validate_header_block(
-    component: RotComponent,
-    slot: SlotId,
-    block: &[u8; BLOCK_SIZE_BYTES],
-) -> Result<(), UpdateError> {
-    let exec = image_range(component, slot).1;
+/// Address ranges that may contain an image during storage and active use.
+/// `store` and `exec` ranges are the same except for `stage0next`.
+pub struct FlashRange {
+    pub store: Range<u32>,
+    pub exec: Range<u32>,
+}
+
+/// Get the flash storage address range and flash execution address range.
+pub fn flash_range(component: RotComponent, slot: SlotId) -> FlashRange {
+    // Safety: Linker defined symbols are trusted.
```

Comment:
How about this for a safety comment?

```rust
// Safety: this block requires unsafe code to generate references to the extern "C" statics.
// Because we're only getting their addresses (all operations below are just as_ptr),
// we can't really trigger any UB here. The addresses themselves are assumed to
// be valid because they're produced by the linker, which we implicitly trust.
```

Reply:
LGTM